### DISCUSSION TITLE
Production readiness

### DISCUSSION THREAD

**Discussion by runenielsen** [infiniticio/infinitic#183] [Q&A] [answered]
Hi Gilles.

Sorry to bother you again, but I noticed the latest version number of this project is low and below 1, but different projects use version numbers very differently.

Could you elaborate on the state the project is in? Is it production ready, beta or alpha?

Are there any serious outstanding issues to be solved?

Thanks in advance,
Rune

**Comment by geomagilles** [comment, marked as answer]
Hi Rune. Infinitic is already used in production by some companies. Since 0.9 I'm ensuring through tests that each new version is backward compatible (ie. you can upgrade Infinitic without breaking your workflows). I'm not aware of any significant bugs. So I'd say you can use it. The reason why it has not reached the 1.0 mark is mostly because I'd like to add some features (use of stateful pulsar functions, better dashboard, gRPC support...)

**Reply by benjamincox** [reply]
@geomagilles On this topic I've had a few concerns myself. However, the 0.15.0 is the first release I've seen that comes with a warning like:

"Please ensure to terminate all running workers prior to upgrading to version 0.15.0. This is crucial because prior versions earlier than 0.15.0, are unable to deserialize messages produced by the new version 0.15.0."

This seems to break backward compatibility, no? 

Is there a suggested approach for those who have many workflows "in flight" at any given time for migrating to this version? I presume most projects can't simply drop their customer's long-running workflows on the floor. 

Feels like I may just be missing something, so I apologize if I've got the wrong impression. 

Thanks for any insight you can provide!

**Reply by geomagilles** [reply]
Hi,

Thank you for reaching out. I'd like to clarify a few points:

-  **Compatibility Issue**: This is actually a forward compatibility issue, not a backward compatibility one. The new version (0.15.0) produces messages that older versions can't understand, but it can still work with messages from previous versions.
-  **Upgrade Process**: To ensure a smooth transition to version 0.15.0, avoid running workers with version 0.15.0 alongside workers with previous versions:
      - Terminate all currently running workers
      - Deploy the upgraded version (0.15.0) across your system
- **Long-running Workflows**: Rest assured, your existing long-running workflows will not be impacted by this upgrade.

I hope this addresses your concerns and provides a clear path forward for
upgrading. If you have any further questions or need additional clarification, please don't hesitate to ask.

Best

**Comment by runenielsen** [comment]
Thanks again! 🙂
